include/xsimd/types/xsimd_register.hpp

```diff
@@ -20,7 +20,9 @@ namespace xsimd
     {
 
         template <class T, class Arch>
-        struct simd_register;
+        struct simd_register
```

Comment:
I <3 all of your changes. Can you explain that one?

Reply:
Ha, yes. Without that, the `static_assert` in `batch` never fires, because `simd_register` is not defined for unsupported architectures (there's only a forward declaration, which can't be inherited from).

That's why i'm not sure about it, as it means that `simd_register` can be constructed for unsupported architectures. I couldn't think of any way around this except for making it's constructor private, but that feels messy.

Reply:
What about putting the same static_assert in `simd_register` then?

Reply:
Oh yeah, that works. I tried this originally but got confused.

The difference now from the original is if you try to construct a batch with a bad type; you used to get

```
invalid use of incomplete type ‘struct xsimd::types::simd_register<double*, xsimd::sse2>’
```

whereas now you get

```
no type named ‘register_type’ in ‘struct xsimd::types::simd_register<double*, xsimd::sse2>’
```
I think it's fine; `simd_register` is presumably an internal thing anyway.